regulation-worker/internal/delete/api/api.go

```diff
@@ -41,9 +41,8 @@ func (*APIManager) GetSupportedDestinations() []string {
 	return supportedDestinations
 }
 
-// prepares payload based on (job,destDetail) & make an API call to transformer.
-// gets (status, failure_reason) which is converted to appropriate model.Error & returned to caller.
-func (api *APIManager) Delete(ctx context.Context, job model.Job, destination model.Destination) model.JobStatus {
+func (api *APIManager) delete(ctx context.Context, job model.Job, destination model.Destination, attempts int) model.JobStatus {
```

Comment:
`Delete` and `delete` names for the functions would be very confusing. Please pick something else.
Suggestion: Call the same method recursively as we have a terminating condition now.

Reply:
we need another function as golang doesn't support default or optional params

Reply:
Ok, then 'Delete' and 'DeleteWithRetyAttempts'.

Reply:
Done